baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/ConjureUnionExhaustiveSwitchTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+public class ConjureUnionExhaustiveSwitchTest {
+
+    @Test
+    void testConjureUnionSwitchWithDefault() {
+        String source = """
+            sealed abstract class Shape permits Circle, Square, Unknown {
+                sealed interface Known permits Circle, Square {}
+            }
+            final class Circle extends Shape implements Shape.Known {}
+            final class Square extends Shape implements Shape.Known {}
+            final class Unknown extends Shape {}
+            public class Test {
+                public void test(Shape shape) {
+                    // BUG: Diagnostic contains: Avoid using default clause
+                    switch (shape) {
+                        case Circle circle -> System.out.println("Circle");
+                        default -> System.out.println("Unknown");
```

Comment:
```suggestion
                        case Circle circle -> System.out.println("Circle");
                        case Square square -> System.out.println("Square");
                        case Unknown unknown -> System.out.println("Unknown");
                        default -> System.out.println("Unknown");
```
Is there a reason not to add all the test cases here? (if we're just testing the behavior with regards to the presence of `default`, this seems more likely to be the test we want - we can also separately test the check behaves correctly in the absence of some statement, should we want to)

Reply:
I believe there's already an existing check that checks for an "unnecessary" default clause if you've written an exhaustive switch statement over an enum/sealed type (not applied in this test, but also not exactly the goal of this check). In this example the default case is necessary since we don't go over all the possible subvariants, but still gets flagged by the check.